components/gfx/platform/freetype/font.rs

```diff
@@ -341,13 +352,9 @@ impl<'a> PlatformFont {
         }
     }
 
-    #[allow(clippy::mut_from_ref)] // Intended for this function
-    fn face_rec_mut(&'a self) -> &'a mut FT_FaceRec {
```

Comment:
I was wondering if it is better to keep `face_rec_mut` and have it defined as:

```
    fn face_rec_mut(&'a self) -> MappedReentrantMutexGuard<'a, FT_FaceRec> {
       ReentrantMutexGuard::map(self.face.lock(), |face| unsafe { &**face })
    }
```

One advantage of doing this would be the `unsafe { **face }` blocks spread across the different methods can be contained within a single method making it easier to audit. Not sure if this makes things easier to abuse in an unsafe way as well.